log/log.go

```diff
@@ -113,13 +149,15 @@ func (l *context) Log(keyvals ...interface{}) error {
 	if len(kvs)%2 != 0 {
 		kvs = append(kvs, ErrMissingValue)
 	}
+	kvs = append(kvs, l.suffixKeyvals...)
 	if l.hasValuer {
-		// If no keyvals were appended above then we must copy l.keyvals so
-		// that future log events will reevaluate the stored Valuers.
+		// If no keyvals were appended above then we must copy l.keyvals
+		// and l.suffixKeyvals so that future log events will reevaluate
+		// the stored Valuers.
 		if len(keyvals) == 0 {
-			kvs = append([]interface{}{}, l.keyvals...)
+			kvs = append([]interface{}{}, append(l.keyvals, l.suffixKeyvals...)...)
 		}
-		bindValues(kvs[:len(l.keyvals)])
+		bindValues(kvs[:(len(kvs))])
```

Comment:
This call to bindValues is not quite what we want.

Before adding `WithSuffix` we only passed in a subslice of `kvs` that included the keyvals stored in `context.keyvals` and excluded the new keyvals passed to the `Log` method.

The new version here includes the new keyvals passed to the `Log` method. I think there should be two calls to `bindValues`, one for the elements of `kvs` from `l.keyvals` and one for the elements from `l.suffixKeyvals`.

We should also consider separate `hasValuer` fields for each group of keyvals stored in the context and handling them separately in the `Log` method.

Reply:
Ah, got a bit confused there. Good spot. Addressed in: https://github.com/go-kit/kit/pull/992/commits/b8ab9fc3b52a767219010f528bed08d306acc918 and https://github.com/go-kit/kit/pull/992/commits/6746fef7864e26bd7399a9630cd09e844c9acda6 by making 2 calls to `bindValues` and creating a separate `hasValuer` for suffix.